### PR TITLE
[feature](nereids)add binary & unary arithmetic for nereids

### PR DESCRIPTION
**Pull request by englefly**
# Proposed changes
 binary arithmetic expression, like div, ^, |, &, %
 unary arithmetic expression, like ~, +1

@morrySnow PTAL
Issue Number: #14075 

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.59 seconds
 load time: 454 seconds
 storage size: 17123356258 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221207025440_clickbench_pr_58983.html

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/ExpressionTranslator.java [hunk 1]
```diff
@@ -346,6 +347,12 @@ public Expr visitBinaryArithmetic(BinaryArithmetic binaryArithmetic, PlanTransla
                 binaryArithmetic.child(1).accept(this, context));
     }
 
+    @Override
+    public Expr visitUnaryArithmetic(UnaryArithmetic unaryArithmetic, PlanTranslatorContext context) {
```
[the remark this answers is not in the document]
Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/ExpressionTranslator.java:351]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/BitNot.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    @Override
+    public <R, C> R accept(ExpressionVisitor<R, C> visitor, C context) {
+        return visitor.visitBitNot(this, context);
+    }
+
+    @Override
+    public AbstractDataType inputType() {
+        return child().getDataType();
```
[the remark this answers is not in the document]
Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/BitNot.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rewrite/rules/TypeCoercion.java [hunk 3]
```diff
@@ -160,6 +164,17 @@ public Expression visitInPredicate(InPredicate inPredicate, ExpressionRewriteCon
                 .orElse(newInPredicate);
     }
 
+    @Override
+    public Expression visitBitNot(BitNot bitNot, ExpressionRewriteContext context) {
+        Expression child = bitNot.child();
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rewrite/rules/TypeCoercion.java:169]: need to rewrite child first.
```
Expression child = rewrite(bitNot.child(), context);
```

Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rewrite/rules/TypeCoercion.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rewrite/rules/TypeCoercion.java [hunk 4]
```diff
@@ -160,6 +164,17 @@ public Expression visitInPredicate(InPredicate inPredicate, ExpressionRewriteCon
...
 
+    @Override
+    public Expression visitBitNot(BitNot bitNot, ExpressionRewriteContext context) {
+        Expression child = rewrite(bitNot.child(), context);
+        if (child.getDataType().toCatalogDataType().getPrimitiveType().ordinal()
+                > PrimitiveType.LARGEINT.ordinal()) {
+            child = new Cast(child, BigIntType.INSTANCE);
+            return bitNot.withChildren(child);
+        }
+        return bitNot;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rewrite/rules/TypeCoercion.java:175]: en..., it is still not right, need to substitute child when child has changed

**Review by morrySnow — APPROVED** [review]

**Pull request merged**